src/constants.ts

```diff
@@ -37,3 +37,26 @@ const PRETTIER_PLUGIN_SORT_IMPORTS_NEW_LINE =
 export const newLineNode = expressionStatement(
     stringLiteral(PRETTIER_PLUGIN_SORT_IMPORTS_NEW_LINE),
 );
+export const injectNewlinesRegex = /"PRETTIER_PLUGIN_SORT_IMPORTS_NEW_LINE";/gi;
+
+/**
+ * If you have a trailing single-line comment on an ImportSpecifier, and attached to
+ *  the following ImportSpecifier you have a leading single-line comment,
+ * Then Babel's code-generator might render the two comments on the same line.
+ *
+ * Additionally, if you have a single-line-comment trailing an ImportDeclaration,
+ *  babel might pull
```

Comment:
Actually, I was able to find https://github.com/babel/babel/pull/14979, which was included in a more recent babel version that we were using.  

I know you put a bunch of thought into this, so I kind of hate to say it, but I updated babel in your branch (and fixed a few minor typescript issues), and this `PATCH_BABEL_GENERATOR_DOUBLE_COMMENTS_ON_ONE_LINE_ISSUE` no longer seems to be needed.  

In fact, it seems like now is a good time to update babel as part of the breaking change, since it may impact the way we do formatting.  I'll push that up as a separate PR, so we can examine any other impacts to existing snapshots.  Then we can apply this branch on top of it, and maybe some of the hoops you had to jump through to get comments working correctly won't be needed anymore.

Sorry I didn't think to upgrade babel sooner.  Because we use a caret in the range, new installations will already be using the new versions.  I start to understand why the upstream project locked their version of babel, but I still think  it's not the best thing to do, and causes more issues than it potentially solves.

Reply:
FWIW, it looks like ASTExplorer is using an older version of babel too.  🤦

Reply:
Yeah, new babel fixes some of the root issues! Removed my workarounds!